tensordict/tensordict.py

```diff
@@ -177,7 +177,7 @@ def _combine_keys(self, prefix, key):
         return key
 
     def __len__(self):
-        return len([key for key in self])
+        return len(list(self))
```

Comment:
I think your errors are coming from this line. I remember running into the same issue, it's because casting something to a `list` calls `__len__` first which leads to a recursion error. The list comprehension doesn't need `__len__`, just calls `__iter__`.

Perhaps we revert this change and add `# noqa` plus an explanation.

Reply:
there's a better option (IMO)
```python
    def __len__(self):
        i = 0
        for _ in self:
            i += 1
        return i
```

like this we're not creating a list for the sole purpose of looking at its length
What do you think

Reply:
Nice, I agree, this is better.